### PR TITLE
Expand mDNS settings

### PR DESCRIPTION
**Pull request by troglobit**
## Description

This PR adds support for modifying the following Avahi daemon settings:

 - `domain-name`
 - `allow-interfaces`
 - `deny-interfaces`
 - `enable-reflector`
 - `reflect-filters`

Also included, a review of the default `sysctl` settings, and some fixes and clarifications to the Networking Guide.

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [X] Feature
  - [X] YANG model change => revision updated?
  - [X] Regression tests added?
  - [x] ChangeLog updates (for next release)
  - [x] Documentation added?
- [ ] Test changes
  - [ ] Checked in changed Readme.adoc (make test-spec)
  - [ ] Added new test to group Readme.adoc and yaml file
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
  - [ ] ChangeLog updated (for major changes)
- [ ] Other (please describe):

### CODE REVIEW

**Comment by troglobit** [comment]
@mattiaswal + @jovatn this PR is now finished and awaiting your review.

**Review by jovatn — APPROVED** [review]
Update https://github.com/kernelkit/infix/blob/main/doc/discovery.md#mdns with (rudimentary) descriptions of available mdns settings.

sysctl updates looks good to me, but perhaps we could have an IRL session for rubber ducking on sysctl changes?

File: src/confd/yang/infix-services.yang [hunk 1]
```diff
@@ -35,6 +46,63 @@ module infix-services {
...
+      type inet:domain-name;
+    }
+
+    container interfaces {
+      description "Filter interfaces to act on.";
+
+      leaf-list allow {
+        description "Enabled on interfaces, can be combined with deny-interfaces.
+
+                     By defaullt all, except loopback and point-to-pint links.";
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/yang/infix-services.yang:65]: Ah, thanks!

File: src/confd/yang/infix-services.yang [hunk 2]
```diff
@@ -35,6 +46,63 @@ module infix-services {
...
+                       - _airplay._tcp.local
+                       - _raop._tcp.local
+
+                     For AirPrint use:
+
+                       - _printer._tcp.local
+                       - _ipp._tcp.local
+                       - _pdl-datastream._tcp.local
+
+                     By defaullt all services are reflected.";
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/yang/infix-services.yang:102]: Thanks!

File: src/confd/yang/infix-services.yang [hunk 3]
```diff
@@ -35,6 +46,63 @@ module infix-services {
...
+        type if:interface-ref;
+      }
+    }
+
+    container reflector {
+      description "Reflect incoming mDNS requests to local interfaces.";
+
+      leaf enabled {
+        description "Enable mDNS reflector on local interfaces.";
+        type boolean;
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/yang/infix-services.yang:85]: The default is disabled.  The rest of the team did not feel the reflector should be enabled by default on all products.

File: doc/ChangeLog.md
Comment (mattiaswal) [inline doc/ChangeLog.md:1]: This should also contain the NTP fixes, merge errors?

File: src/confd/src/infix-services.c [hunk 4]
```diff
@@ -176,21 +176,81 @@ static int hostname_change(sr_session_ctx_t *session, uint32_t sub_id, const cha
...
+}
+
 static void mdns_cname(sr_session_ctx_t *session)
 {
 	int ena = srx_enabled(session, "/infix-services:mdns/enabled");
 
 	if (ena) {
 		int www = srx_enabled(session, "/infix-services:web/netbrowse/enabled");
-		char *name = fgetkey("/etc/os-release", "DEFAULT_HOSTNAME");
+		const char *hostnm = fgetkey("/etc/os-release", "DEFAULT_HOSTNAME");
```
Comment (mattiaswal) [inline src/confd/src/infix-services.c:245]: Use hostname instead of hostnm pleeeeease.

File: test/case/infix_services/Readme.adoc [hunk 5]
```diff
@@ -3,4 +3,5 @@
 
 <<<
 
+include::mdns_allow_deny/Readme.adoc[]
```
Comment (mattiaswal) [inline test/case/infix_services/Readme.adoc:6]: Add a blank line after the include

**Review by mattiaswal — CHANGES_REQUESTED** [review]

File: doc/ChangeLog.md
Comment (troglobit) [reply doc/ChangeLog.md:1]: Ah, yes of course, nice catch!

**Review by mattiaswal — APPROVED** [review]
:100:

**Pull request merged**